src/main/java/com/denizenscript/clientizen/mixin/EntityVisibilityMixin.java

```diff
@@ -0,0 +1,38 @@
+package com.denizenscript.clientizen.mixin;
+
+import com.denizenscript.clientizen.events.EntitySeenUnseenByCamera;
+import com.denizenscript.clientizen.objects.EntityTag;
+import net.minecraft.client.render.Frustum;
+import net.minecraft.client.render.entity.EntityRenderDispatcher;
+import net.minecraft.entity.Entity;
+import net.minecraft.util.math.Box;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Inject;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfoReturnable;
+
+import java.util.List;
+import java.util.UUID;
+
+@Mixin(EntityRenderDispatcher.class)
+public class EntityVisibilityMixin {
+
+    @Inject(method = "shouldRender", at = @At("HEAD"))
+    private <E extends Entity> void clientizen$isEntityVisible(E entity, Frustum frustum, double x, double y, double z, CallbackInfoReturnable<Boolean> cir) {
+        UUID uuid = entity.getUuid();
+        List<UUID> visibleEntities = EntityTag.getVisibleEntities();
+        // Based on net.minecraft.client.render.entity.EntityRenderer#shouldRender
```

Comment:
Is there any reason the logic needs to be copied over? at a glance it looks like it'd be possible to just inject at `RETURN` and grab the boolean the method returns?

Reply:
Oh, I never thought about doing that. I'll take a look into it.